Correct cache eviction policy (#38744)
github/docs: data/reusables/actions/cache-eviction-policy.md

> Once a repository has reached its maximum cache storage, the cache eviction policy will create space by deleting the caches that have been least recently used.

Reviewer: ```suggestion
Once a repository has reached its maximum cache storage, the cache eviction policy will create space by deleting the caches in order of last access date, from oldest to most recent.
```
Wording update to better fit the style guide.
